plugins/templates/createTag.js

```diff
@@ -0,0 +1,84 @@
+'use strict';
+
+const boom = require('boom');
+const joi = require('joi');
+const schema = require('screwdriver-data-schema');
+const baseSchema = schema.models.templateTag.base;
+const urlLib = require('url');
+
+/* Currently, only build scope is allowed to tag template due to security reasons.
+ * The same pipeline that publishes the template has the permission to tag it.
+ */
+module.exports = () => ({
+    method: 'PUT',
+    path: '/templates/{templateName}/tags/{tagName}',
+    config: {
+        description: 'Add or update a template tag',
+        notes: 'Add or update a specific template',
+        tags: ['api', 'templates'],
+        auth: {
+            strategies: ['token', 'session'],
+            scope: ['build']
+        },
+        plugins: {
+            'hapi-swagger': {
+                security: [{ token: [] }]
+            }
+        },
+        handler: (request, reply) => {
+            const pipelineFactory = request.server.app.pipelineFactory;
+            const templateFactory = request.server.app.templateFactory;
+            const templateTagFactory = request.server.app.templateTagFactory;
+            const pipelineId = request.auth.credentials.pipelineId;
+            const name = request.params.templateName;
+            const tag = request.params.tagName;
+            const version = request.payload.version;
+
+            return Promise.all([
+                pipelineFactory.get(pipelineId),
+                templateFactory.get({ name, version }),
```

Comment:
Should be `get`, since we want to get the exact version. And `getTemplate` will be refactored into string only later.